pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -51,110 +71,54 @@ pub mod pallet {
 		#[pallet::weight(0)]
 		pub fn initialize_reward(
 			origin: OriginFor<T>,
-			contributions: Vec<(T::RelayChainAccountId, u32)>,
-			rate: u32,
+			contributions: Vec<(T::AccountId, BalanceOf<T>)>,
 			end_block: T::BlockNumber,
 		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
+
 			let now = frame_system::Pallet::<T>::block_number();
```

Comment:
I think will much correct if we use a relay chain block instead of parachain block.
https://github.com/paritytech/polkadot/blob/master/primitives/src/v1/mod.rs#L439.
We can do something like:
```
				let relay_block = cumulus_pallet_parachain_system::Pallet::<T>::validation_data()
					.expect("validation data was set in parachain system inherent")
					.relay_parent_number;

```

Reply:
Use the relay chain block to calculate reward:
https://github.com/polkafoundry/newpolka/pull/55/commits/c67c885a9d1f03eaa8f958b24750eee4a6531327#diff-2bbae40c9c7fdfeeb0b1708d368844303a595083bc948ccd684b577e5d13f6ebR105